flake.nix

```diff
@@ -255,9 +245,10 @@
                 # Combined project coverage report
                 testCoverageReport = coveredProject.projectCoverageReport;
                 # `tests` are the test suites which have been built.
-                tests = removeRecurse (collectComponents "tests" isProjectPackage coveredProject.hsPkgs);
+                tests =
+                  lib.removeRecurse (collectComponents "tests" isProjectPackage coveredProject.hsPkgs);
```

Comment:
I see some minor repetition between this and 2 expressions below, maybe its worth factoring it out into another library function `collectProjectComponents` ?

then it could be applied like this:
`collectProjectComponents "tests" coveredProject`
`collectProjectComponents "benchmarks" project`

Reply:
Perhaps, but I would consider this out of scope for this pull request.

The function `removeRecurse` is sufficiently obscure that I don't quite understand why it is called in the first case, and why `collectComponents` by itself does not suffice. 🤔